### ISSUE TITLE
Set Creation: New Problem

### ISSUE THREAD

**Issue by alexrcabral** [benetech/MathShare#1601]
Set Creation: New Problem ((Plus, no wrapping)) 

<img width="498" alt="Mockup of problem set creation on a mobile view. Below on the left is a back arrow to go back to the dashboard, and on the right is a button with a share icon which says Assign Set. Beneath this is a banner with the title New Problem Set 12-20-2020 with a pencil icon. Below this are boxed numbers, 02 through 06. 04 is currently highlighted. There is a plus button on the right hand side of this line. There are left and right arrows on this same line, on either side of the numbered boxes. In the main portion of the screen and taking up the majority of the page is a box with two sections, titled, 'Type the problem here' and 'Add the prompt and any details here'. At the top of the screen is the universal nav bar with the Mathshare logo and the profile dropdown" src="https://user-images.githubusercontent.com/47576069/102955329-80157e80-448a-11eb-8122-48318dcea918.png">

## Add New Problem Button 

On the right hand side of the line with the numbered boxes should be a button to add a new problem to the set. This is represented by a plus icon within a circle.

[white plus icon over a dark purple circle] 

Alt Text: "Add New Problem" 

When clicked, this should save any progress to the current problem & add a new problem at the end of the set. This new problem should become the active problem being edited.

## Notable Changes from Prior Mockups

One change to note from the earlier problem set creation GitHub issues: the arrows should be located on either side of the numbered boxes rather than on the right hand side of the screen. There is also the addition of a plus button for adding new problems.

The colors used for the font and the background for the numbered navigation boxes are different from Eric's design in Invision. The original combination was not high enough contrast to be accessible for low vision users. 

## References 

Style Guides: https://invis.io/CQYLLW8J49U
Mobile Mockups: https://invis.io/5HYLLW1G4BD
Fonts: https://www.dropbox.com/sh/c12m6ufweyds20d/AAAdXNPOt8e52DkSAr9pd-a-a/Fonts?dl=0

## Acceptance criteria: 

Given that I'm a user on v2 who has created or is editing a problem set, I should encounter the "Add New Problem" button to the right of the problem navigation. When I click or select this button, any progress I've made should be saved, and a new problem should be added to the end of the set. This new problem should become my active problem being edited.

**Comment by abaranowska1** [comment]
**Tested with success**
- Given that I'm a user on v2 who has created or is editing a problem set, I encounter the "Add New Problem" button to the right of the problem navigation. 
- When I click or select this button, any progress I've made is saved, and a new problem should be added to the end of the set.
- Alt Text: "Add New Problem" 
![Selection_934](https://user-images.githubusercontent.com/61964176/105042328-66089500-5a64-11eb-99d3-51c72019b156.png)
![Selection_935](https://user-images.githubusercontent.com/61964176/105042338-6739c200-5a64-11eb-83ef-8a0944c5f344.png)

**Comment by alexrcabral** [comment]
This differs from scope on one aspect -- currently when I click the new problem button, it will only add if there is content added to the last problem. This should not be restricted (e.g. for the use case of pre-populating the number of problems you intend to have in the set). 

Next Step: Add step should be possible even when the most recent step is blank

**Comment by johnhbenetech** [comment]
@alexrcabral should we allow for the saving of a blank problem?

**Comment by alexrcabral** [comment]
No, not in the sense of saving a blank problem indefinitely for return to problem creation later.

For the duration of the problem creation session it needs to be able to handle multiple blank problems. If a problem is left empty when the user leaves problem set creation, that no longer needs to be retained.

**Comment by johnhbenetech** [comment]
@alexrcabral please keep in mind that since there is no longer a save button, we save based on actions taken in the interface. 

So if a user has a blank problem 2 and then adds a problem 3 with content, what happens from there? If the user closes the page what should happen? If the user assigns the set in this state what happens? If a user comes back to the edit view what happens?

**Comment by johnhbenetech** [comment]
Updated task: Add problem button to be grayed out if user is on last problem and has not entered any data.

**Comment by abaranowska1** [comment]
**Tested with success**
The add problem icon/button is grayed out if the user is on the last problem and has not entered any data.
![image](https://user-images.githubusercontent.com/61964176/107334341-e3995100-6ab6-11eb-9680-6fe4f4597f32.png)
After going back and open the problem once again, the user sees only before-filled problems.
![image](https://user-images.githubusercontent.com/61964176/107334479-0e83a500-6ab7-11eb-9aab-06a343dce466.png)
